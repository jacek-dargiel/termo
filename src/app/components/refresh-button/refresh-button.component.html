<button
  class="refresh"
  (click)="onRefreshClick()"
>
  <div
    class="refresh__progress"
    [ngClass]="{ 'refresh__progress--refreshing': refreshing }"
  ></div>
  <span class="refresh__label"><ng-content></ng-content></span>
</button>
