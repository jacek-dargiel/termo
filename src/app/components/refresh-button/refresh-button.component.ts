import { Component, input, output } from '@angular/core';
import { NgClass } from '@angular/common';

@Component({
    selector: 'termo-refresh-button',
    templateUrl: './refresh-button.component.html',
    styleUrls: ['./refresh-button.component.scss'],
    imports: [NgClass]
})
export class RefreshButtonComponent {
  readonly progress = input<number>();
  readonly refreshing = input<boolean>();
  readonly refresh = output();

  onRefreshClick() {
    this.refresh.emit();
  }
}
