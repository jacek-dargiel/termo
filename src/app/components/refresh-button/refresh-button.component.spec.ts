import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { RefreshButtonComponent } from './refresh-button.component';

describe('RefreshButtonComponent', () => {
  let component: RefreshButtonComponent;
  let fixture: ComponentFixture<RefreshButtonComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ RefreshButtonComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(RefreshButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render button at initial state', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should render button at 50%', () => {
    component.progress = 0.5;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should render button at refreshing state', () => {
    component.progress = 1;
    component.refreshing = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
