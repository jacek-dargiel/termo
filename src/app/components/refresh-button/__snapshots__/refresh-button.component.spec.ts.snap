// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RefreshButtonComponent should render button at 50% 1`] = `
<termo-refresh-button
  progress={[Function Number]}
  refresh={[Function EventEmitter_]}
>
  <button
    class="refresh"
  >
    <div
      class="refresh__progress"
      style="width: 50%;"
    />
    <span
      class="refresh__label"
    />
  </button>
</termo-refresh-button>
`;

exports[`RefreshButtonComponent should render button at initial state 1`] = `
<termo-refresh-button
  refresh={[Function EventEmitter_]}
>
  <button
    class="refresh"
  >
    <div
      class="refresh__progress"
    />
    <span
      class="refresh__label"
    />
  </button>
</termo-refresh-button>
`;

exports[`RefreshButtonComponent should render button at refreshing state 1`] = `
<termo-refresh-button
  progress={[Function Number]}
  refresh={[Function EventEmitter_]}
  refreshing={[Function Boolean]}
>
  <button
    class="refresh"
  >
    <div
      class="refresh__progress refresh__progress--refreshing"
      style="width: 100%;"
    />
    <span
      class="refresh__label"
    />
  </button>
</termo-refresh-button>
`;
